### PR TITLE
[BEAM-1651] Add code style xml to the project repository

### PR DESCRIPTION
**Pull request by aviemzur**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by aviemzur** [comment]
R: @jbonofre

**Comment by jbonofre** [comment]
I gonna merge.

**Comment by aviemzur** [comment]
@jbonofre working on an improvement discussed with @davorbonaci in the JIRA ticket.

**Comment by aviemzur** [comment]
@jbonofre PTAL

Changed to use Eclipse code formatter profile XML which can be used in both IntelliJ and Eclipse.
I generated it using Eclipse checkstyle plugin which creates a code formatter profile based on a checkstyle xml (I used our checkstyle xml).
I added a plugin which formats a test class using the xml before checkstyle is run to test the code style xml is correct.

CC: @davorbonaci

**Comment by aviemzur** [comment]
Retest this please.

**Comment by jbonofre** [comment]
Thanks, let me take a new look.

**Comment by kennknowles** [comment]
Noting that there is already a `google-java-format` plugin for IntelliJ which works really really well.

File: sdks/java/build-tools/pom.xml [hunk 1]
```diff
@@ -28,5 +28,34 @@
...
+                <directory>${project.build.testSourceDirectory}</directory>
+              </directories>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-checkstyle-plugin</artifactId>
```
Comment (davorbonaci) [inline sdks/java/build-tools/pom.xml]: Isn't this inherited?

File: sdks/java/build-tools/pom.xml [hunk 2]
```diff
@@ -28,5 +28,34 @@
 
   <artifactId>beam-sdks-java-build-tools</artifactId>
   <name>Apache Beam :: SDKs :: Java :: Build Tools</name>
-  
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>net.revelc.code.formatter</groupId>
+        <artifactId>formatter-maven-plugin</artifactId>
```
Comment (davorbonaci) [inline sdks/java/build-tools/pom.xml]: License analysis: ok. No action necessary.

File: sdks/java/build-tools/pom.xml (hunk 1 above)
Comment (aviemzur) [inline sdks/java/build-tools/pom.xml]: `build-tools` parent is `beam-parent`. So `maven-checkstyle-plugin` is inherited in `pluginsManagement` but not in `plugins`.

**Comment by aviemzur** [comment]
We should add a link to `google-java-format` plugin in the IntelliJ and Eclipse sections on the website.
Do we still want this XML as an alternative, or is the plugin sufficient?

**Comment by davorbonaci** [comment]
Merging.

**Pull request closed without merging**